mp/src/game/client/momentum/ui/HUD/hud_stickybombcharge.cpp

```diff
@@ -0,0 +1,124 @@
+#include "cbase.h"
+#include <vgui/ILocalize.h>
+#include <vgui/ISurface.h>
+#include <vgui/IVGui.h>
+#include <vgui_controls/EditablePanel.h>
+#include <vgui_controls/ProgressBar.h>
+#include <vgui_controls/Label.h>
+#include "c_mom_player.h"
+#include "hud.h"
+#include "hudelement.h"
+#include "iclientmode.h"
+#include "ienginevgui.h"
+#include "mom_system_gamemode.h"
+#include "weapon/weapon_mom_stickybomblauncher.h"
+
+#include "tier0/memdbgon.h"
+
+using namespace vgui;
+
+static MAKE_TOGGLE_CONVAR(mom_hud_sj_chargemeter_enable, "1", FCVAR_ARCHIVE, "Toggles the charge meter on or off.\n");
+
+class CHudStickyCharge : public CHudElement, public EditablePanel
+{
+    DECLARE_CLASS_SIMPLE(CHudStickyCharge, EditablePanel);
+
+  public:
+    CHudStickyCharge(const char *pElementName);
+
+    bool ShouldDraw() OVERRIDE;
+    void OnThink() OVERRIDE;
+    void Reset() OVERRIDE;
+
+  private:
+    CMomentumStickybombLauncher *m_pLauncher;
+    ContinuousProgressBar *m_pChargeMeter;
+    Label *m_pChargeLabel;
+};
+
+DECLARE_HUDELEMENT(CHudStickyCharge);
+
+CHudStickyCharge::CHudStickyCharge(const char *pElementName)
+    : CHudElement(pElementName), EditablePanel(g_pClientMode->GetViewport(), "HudStickyCharge")
+{
+    m_pLauncher = nullptr;
+    m_pChargeMeter = new ContinuousProgressBar(this, "ChargeMeter");
+    m_pChargeLabel = new Label(this, "ChargeMeterLabel", "CHARGE");
+
+    LoadControlSettings("resource/ui/HudStickyCharge.res");
+}
+
+void CHudStickyCharge::Reset()
+{
+    m_pLauncher = nullptr;
+}
+
+bool CHudStickyCharge::ShouldDraw()
+{
+    static ConVarRef mom_sj_charge_enable("mom_sj_charge_enable");
+
+    if (!mom_hud_sj_chargemeter_enable.GetBool())
+        return false;
+
+    C_MomentumPlayer *pPlayer = C_MomentumPlayer::GetLocalMomPlayer();
+
+    if (!pPlayer || !g_pGameModeSystem->GameModeIs(GAMEMODE_SJ) || !pPlayer->IsAlive())
+        return false;
+
+    const auto pWeapon = pPlayer->GetActiveWeapon();
+
+    if (!pWeapon)
+        return false;
+
+    if (pWeapon->GetWeaponID() != WEAPON_STICKYLAUNCHER)
+    {
+        return false;
+    }
+    else
+    {
+        if (!m_pLauncher)
+            m_pLauncher = static_cast<CMomentumStickybombLauncher *>(pWeapon);
+
+        return true;
```

Comment:
Upon looking at the rest of the function, you'll want to fall through to the CHudElement return, so you can remove this return and be alright.